karapace/in_memory_database.py

```diff
@@ -217,19 +217,19 @@ def find_subjects(self, *, include_deleted: bool) -> list[Subject]:
                 subject for subject in self.subjects if self.find_subject_schemas(subject=subject, include_deleted=False)
             ]
 
-    def find_subject_schemas(self, *, subject: Subject, include_deleted: bool) -> dict[ResolvedVersion, SchemaVersion]:
+    def find_subject_schemas(self, *, subject: Subject, include_deleted: bool) -> dict[Version, SchemaVersion]:
         if subject not in self.subjects:
             return {}
         if include_deleted:
             return self.subjects[subject].schemas
         with self.schema_lock_thread:
             return {
-                version_id: schema_version
+                Versioner.V(version_id): schema_version
```

Comment:
I'd give it a closer look, I am not sure tho, as you've right mentioned, this data is read and set on the class attribute - `InMemoryDatabase.subjects` which is inserted via `database.insert_schema_version()` and then this is called on the `KafkaSchemaReader._handle_msg_schema()`, which is called on `KafkaSchemaReader.handle_msg()`, which is called from `KafkaSchemaReader.handle_messages()` which operates on a `confluent_kafka.Message().value` and thus I do not see any explicit creation of the `Version` anywhere up that chain. It's either I perform this here or we move it closer to `KafkaSchemaReader._handle_msg_schema()` which I didn't do, so as to avoid touching the reader logic.

Reply:
I'd leave as is, wdyt?

Reply:
Leave as is, `schema_reader` requires a fix to use `Version` and not raw `int`.